### PR TITLE
Fix fundslocker to use Task API properly

### PR DESCRIPTION
**Pull request by mplebanski**
This fixes fundslocker to use appropriate API method from Task class instead
of accessing property that is only defined in a CoreTask (Task
derived class). Task base class does not have the `total_tasks
property. This code caused problems when deriving Task from scratch.

### CODE REVIEW

File: tests/golem/ethereum/test_fundslocker.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def make_mock_task(*_, task_id: str = 'tid', subtask_price: int = 100,
     task.header.deadline = timeout_to_deadline(timeout)
     task.header.task_id = task_id
     task.subtask_price = subtask_price
-    task.total_tasks = total_tasks
+    task.get_total_tasks = mock.Mock(return_value=total_tasks)
```
Comment (Krigpl) [inline tests/golem/ethereum/test_fundslocker.py]: or
`task.get_total_tasks.return_value = total_tasks`

**Review by Krigpl — APPROVED** [review]

File: tests/golem/test_client.py [hunk 2]
```diff
@@ -810,8 +810,8 @@ def test_enqueue_new_task_from_type(self, *_):
         task_mock = MagicMock()
         task_mock.header.max_price = 1 * 10**18
         task_mock.header.subtask_timeout = 158
-        task_mock.total_tasks = 3
-        price = task_mock.header.max_price * task_mock.total_tasks
+        task_mock.get_total_tasks = mock.Mock(return_value=3)
```
Comment (etam) [inline tests/golem/test_client.py]:
```suggestion
        task_mock.get_total_tasks.return_value=3
```

**Review by etam — APPROVED** [review]
ok

**Comment by mplebanski** [comment]
Thanks for the comments

**Pull request merged**